### ISSUE TITLE
Implement release channels for CLI

### ISSUE THREAD

**Issue by maroshii** [okteto/okteto#2738]
**TL;DR** - This is effectively implementing  the release doc added in #2737.

---

Releasing new versions of Okteto will consist of releasing binaries through three major channels: **dev, beta and stable.**

The dev channel builds nightly from master and doesn't follow semver. 
The beta channel automatically builds from `release-MAJOR.MINOR` branches (semver with `-beta.n` prerelease)
The stable channel is built from tags manually created. (semver:  MAJOR.MINOR.PATCH)

- [x] CI fixes - #2727 
- [x] Beta Channel - #2727 
- [x] Stable Channel - #2727 
- [x] Deprecate beta.okteto.com - #2727 
- [x] Automatically release for lts branches - #2727 
- [x] New artifact storage mechanism (in downloads.okteto.com) - #2727 
- [x] Update get.okteto.com - #2727 
- [x] Dev Channel
